src/nvim/mapping.c

```diff
@@ -802,8 +814,16 @@ static int buf_do_map(int maptype, MapArguments *args, int mode, bool is_abbrev,
     }
 
     mp->m_keys = vim_strsave(lhs);
-    mp->m_str = vim_strsave(rhs);
-    mp->m_orig_str = vim_strsave(orig_rhs);
+    if (keyround1_simplified) {
+      // ugh! need a second copy!
+      mp->m_str = vim_strsave(args->rhs);
+      mp->m_orig_str = vim_strsave(args->orig_rhs);
```

Comment:
I think this copy can be avoided if these are not freed when unmapping a mapping with `m_simplified`, just like how it is currently done for RHS luarefs.